### PR TITLE
document dynamic primitive (erc-721)

### PR DESCRIPTION
**Pull request by ecioppettini**
documentation for https://github.com/PaimaStudios/paima-engine/pull/362

### CODE REVIEW

File: docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Dynamic ERC721
```
Comment (SebastienGllmt) [inline docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:1]: I feel this is a conceptually complex feature with multiple subtle things to take into account and I feel the docs are too short to properly be addressing all of these. Additionally, it's not clear from the docs how you would even write a smart contract in a way to trigger dynamic primitive so we should probably provide a concrete example of this and discuss best practices like factory contracts

File: docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md [hunk 2]
```diff
@@ -12,35 +18,74 @@ extensions:
...
+Generated extensions have a _name_, that is derived from the name of the trigger
+extension, and the order of the dynamic primitive. The following function can be
+used to generate this name.
+
+```ts
+export function generateDynamicPrimitiveName(parentName: string, id: number): string;
+```
+
+For example, the first dynamic primitive that gets created will have a name of
+`generateDynamicPrimitiveName(parentCdeName, 0)`, and so on. This name can be
```
Comment (SebastienGllmt) [inline docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:76]:
```suggestion
`generateDynamicPrimitiveName(parentPrimitiveName, 0)`, and so on. This name can be
```

File: docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md [hunk 3]
```diff
@@ -12,35 +18,74 @@ extensions:
...
+- `scheduledPrefix` and `burnScheduledPrefix` are the prefixes used for the
+resulting ERC721 events. This primitive won't emit events by itself. See [the
+ERC721 section for details](ERC721#meaning).
+
+#### Dynamic configuration
+
+- `contractAddress` has the name of the field in the emitted event that contains
+the address of the ERC721.
 
 The example configuration assumes that the event has the following signature:
```
[the remark this answers is not in the document]
Comment (SebastienGllmt) [reply docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:59]: We should also make it very clear whether or not the event emission can contain more parameters than needed, and if the position of the argument matters

File: docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md [hunk 4]
```diff
@@ -12,35 +18,74 @@ extensions:
...
 
 ```solidity
-event CustomEvent(address indexed address);
+event CustomEvent(address indexed nftAddress);
```
[the remark this answers is not in the document]
Comment (SebastienGllmt) [reply docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:62]: If we want a full example of a Solidity contract for this, we can use

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.20;

import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";

contract CustomERC721 is ERC721 {
    constructor(string memory name, string memory symbol) ERC721(name, symbol) {}
}

contract FactoryERC721 {
    event ERC721Created(address indexed nftAddress); // emitted when ERC721 token is deployed

    /// @dev deploys an ERC721 token with given parameters
    /// @return address deployed
    function deployERC721(string memory _name, string memory symbol) public returns (address) {
        CustomERC721 t = new CustomERC721(_name, symbol);
        emit ERC721Created(address(t));
        return address(t);
    }
}
```

File: docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md [hunk 5]
```diff
@@ -12,35 +18,74 @@ extensions:
...
+as json. It can be used to extract, the contract address, for example.
+
+```ts
+export async function getDynamicExtensions(
+  readonlyDBConn: Pool,
+  parent: string
+): Promise<{ name: string; config: string }[]>;
+```
 
 ## Performance implications
```
[the remark this answers is not in the document]
Comment (ecioppettini) [reply docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:90]: > the rollup monitors

why rollup? what are we using that term for? I'm not sure I get it.

[the remark this answers is not in the document]
Comment (SebastienGllmt) [reply docs/home/300-react-to-events/2-primitive-catalogue/10-evm/5-dynamic-erc721.md:90]: Your game is a rollup that monitors multiple primitives
https://blog.paimastudios.com/self-sovereign-rollup

**Pull request merged**